nexus/db-queries/src/db/datastore/db_metadata.rs

```diff
@@ -324,44 +390,68 @@ impl DataStore {
             // Now that the schema change has completed, set the following:
             // - db_metadata.version = new version
             // - db_metadata.target_version = NULL
-            self.finalize_schema_update(&current_version, &target_version)
-                .await
-                .map_err(|e| e.to_string())?;
+            //
+            // The "version" field ignores all this pre-release versioning that we
+            // abuse in the "target_version" field.
```

Comment:
At the top of this loop, should we check and bail out if the incoming `target_version` already has a prerelease tag? I don't think anyone is likely to put a prerelease tag on a schema migration version, but if they did it definitely wouldn't behave as they'd expect after these changes.

Reply:
Good call, I'll do this to be defensive.